src/java/org/apache/cassandra/service/MigrationManager.java

```diff
@@ -188,6 +207,24 @@ public static boolean shouldPullSchemaFrom(InetAddress endpoint)
                 && !Gossiper.instance.isGossipOnlyMember(endpoint);
     }
 
+    public static boolean shouldPullSchemaFrom(InetAddress endpoint, UUID theirVersion)
+    {
+        /*
+         * Don't request schema from nodes with a differnt or unknonw major version (may have incompatible schema)
+         * Don't request schema from fat clients
+         * Don't request schema from bootstrapping nodes (?)
+         * Don't request schema if we have scheduled a pull request for that schema version
+         */
+        boolean isOtherSchemaNonEmpty = !Schema.emptyVersion.equals(theirVersion);
```

Comment:
This is an additional check to prevent making schema pull requests to new nodes. It's probably not that important, since the work done in response to that is pretty small.